### PR TITLE
Ll insertions

### PR DESCRIPTION
**Pull request by MerrybyPractice**

### CODE REVIEW

**Comment by MerrybyPractice** [comment]
- [x] .append(value) which adds a new node with the given value to the end of the list
- [x] .insertBefore(value, newVal) which add a new node with the given newValue immediately before the first value node
- [x] .insertAfter(value, newVal) which add a new node with the given newValue immediately after the first value node
Testing 

- [x] Can successfully add a node to the end of the linked list
- [x] Can successfully add multiple nodes to the end of a linked list
- [x] Can successfully insert a node before a node located i the middle of a linked list
- [ ] Can successfully insert a node before the first node of a linked list
- [x] Can successfully insert after a node in the middle of the linked list
- [x] Can successfully insert a node after the last node of the linked list

File: Data-Structures/LinkedList/src/main/java/LinkedList/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        while (current.reference != null) {
+            current = current.reference;
+        }
+
+        NodeConstructor<generic> newNode = new NodeConstructor<>(value, null);
+
+        try {
+            current.reference = newNode;
+        } catch (NullPointerException ex) {
+            System.out.println("I see that you are trying to append a node to an empty list. Instead of using the append method for this operation, I would recommend you try the insert method instead, to create an initial head. I am sure you will find that it meets your needs more thoroughly than append in this case. Once you have a single node created, you are able to use append with abandon.");
```
Comment (JessLovell) [inline Data-Structures/LinkedList/src/main/java/LinkedList/LinkedList.java:65]: This is a long error message. Something you could do instead: have the method return a boolean if the node was not appended. ORR set the head of the list to the desired node.

File: Data-Structures/LinkedList/src/test/java/LinkedList/AppTest.java [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+import org.junit.Test;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+
+import static org.junit.Assert.*;
+
+public class AppTest {
+
+    //creating global variables to be used in subsequent testing. Strictly for ease of writing.
```
Comment (JessLovell) [inline Data-Structures/LinkedList/src/test/java/LinkedList/AppTest.java:15]: Cool. I accept this.

File: Data-Structures/LinkedList/src/test/java/LinkedList/AppTest.java [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+import org.junit.Test;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+
+import static org.junit.Assert.*;
+
+public class AppTest {
+
```
Comment (JessLovell) [inline Data-Structures/LinkedList/src/test/java/LinkedList/AppTest.java:14]: There are a lot of comments in this file.

**Pull request merged**